hudi-flink/pom.xml

```diff
@@ -124,28 +124,77 @@
       <artifactId>kafka-clients</artifactId>
       <version>${kafka.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-hadoop-compatibility_${scala.binary.version}</artifactId>
+      <version>${flink.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-avro</artifactId>
+      <version>${flink.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-json</artifactId>
+      <version>${flink.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-table-common</artifactId>
+      <version>${flink.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-table-runtime-blink_${scala.binary.version}</artifactId>
```

Comment:
I am thinking about one question: how to serve the users who only use DataStream API. Only define one `${flink.version}`? Here introduced some artifacts which only created in the high-level flink version.

Reply:
That's why i mark the dependency as provided, so that the artifacts are not packaged into the final release, low-level flink version users should not be affected.

Reply:
I mean if users only change the Flink version placeholder(`${flink.version}`), it will still cause the compile error, right? Maybe we need to use `<profile>` for the bundle module or another solution?

Reply:
I think we should not package the in-compatible Flink jars into the bundle, just like the guideline [1], the hudi bundle is used as a spark shell library.

The table jars introduced here like `flink-table-common` and `flink-table-runtime-blink` are only used by the new pipeline, so we can package the into the `hoodie-flink` jar, the old pipeline should not be affected. The data stream api should be compatible itself, if the old pipeline does not use API comes from 1.10+ version.

[1] https://hudi.apache.org/docs/quick-start-guide.html